src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -632,21 +642,45 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
         ᶜS_q_totʲ = ᶜS_q_totʲs.:($j)
         ᶜS_e_totʲ_helper = ᶜS_e_totʲs_helper.:($j)
 
-        u³ʲ_halflevel = Fields.field_values(Fields.level(ᶠu³ʲ, i_top + half))
+        u³ʲ_halflevel = F.field_values(F.level(ᶠu³ʲ, i_top + half))
         @. u³ʲ_halflevel = CT3(0)
 
-        entrʲ_level = Fields.field_values(Fields.level(ᶜentrʲ, i_top))
-        detrʲ_level = Fields.field_values(Fields.level(ᶜdetrʲ, i_top))
+        entrʲ_level = F.field_values(F.level(ᶜentrʲ, i_top))
+        detrʲ_level = F.field_values(F.level(ᶜdetrʲ, i_top))
         fill!(entrʲ_level, RecursiveApply.rzero(eltype(entrʲ_level)))
         fill!(detrʲ_level, RecursiveApply.rzero(eltype(detrʲ_level)))
         @. ᶜuʲ = C123(Y.c.uₕ) + ᶜinterp(C123(ᶠu³ʲ))
 
-        S_q_totʲ_level = Fields.field_values(Fields.level(ᶜS_q_totʲ, i_top))
-        S_e_totʲ_helper_level =
-            Fields.field_values(Fields.level(ᶜS_e_totʲ_helper, i_top))
+        S_q_totʲ_level = F.field_values(F.level(ᶜS_q_totʲ, i_top))
+        S_e_totʲ_helper_level = F.field_values(F.level(ᶜS_e_totʲ_helper, i_top))
         @. S_q_totʲ_level = 0
         @. S_e_totʲ_helper_level = 0
     end
+    return nothing
+end
+
+"""
+    set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
+
+Updates the precomputed quantities stored in `p` for diagnostic edmfx.
+"""
+function set_diagnostic_edmf_precomputed_quantities_closures!(Y, p, t)
```

Comment:
(Just want to note that it's more accurate to call it something like environment closures as the closures related to updraft, including entrainment, detrainment, nonhydrostatic pressure drag, and microphysics are in the integral. But I'm fine with the current name.)